account_move_line_product/views/account_move_line_view.xml

```diff
@@ -0,0 +1,76 @@
+<openerp>
+    <data>
+
+
+        <record id="view_move_line_form" model="ir.ui.view">
+            <field name="name">account.move.line.form</field>
+            <field name="model">account.move.line</field>
+            <field name="inherit_id" ref="account.view_move_line_form"/>
+            <field name="arch" type="xml">
+                <field name="quantity" position="before">
+                    <field name="product_id"/>
+                </field>
+                <field name="quantity" position="attributes">
+                    <attribute name="invisible">False</attribute>
+                </field>
+
+            </field>
+        </record>
+
+        <record id="view_move_line_form2" model="ir.ui.view">
+            <field name="name">account.move.line.form2</field>
+            <field name="model">account.move.line</field>
+            <field name="inherit_id" ref="account.view_move_line_form2"/>
+            <field name="arch" type="xml">
+                <field name="quantity" position="before">
+                    <field name="product_id"/>
+                </field>
+                <field name="quantity" position="attributes">
+                    <attribute name="invisible">False</attribute>
+                </field>
+            </field>
+        </record>
+
+        <record id="view_move_line_tree" model="ir.ui.view">
+            <field name="name">account.move.line.tree</field>
+            <field name="model">account.move.line</field>
+            <field name="inherit_id" ref="account.view_move_line_tree"/>
+            <field name="arch" type="xml">
+                <field name="partner_id" position="after">
+                    <field name="product_id"/>
+                    <field name="quantity"/>
+                </field>
+            </field>
+        </record>
+
+        <record id="view_account_move_line_filter" model="ir.ui.view">
+            <field name="name">Journal Items</field>
+            <field name="model">account.move.line</field>
+            <field name="inherit_id" ref="account.view_account_move_line_filter"/>
+            <field name="arch" type="xml">
+                <field name="partner_id" position="after">
+                    <field name="product_id"/>
+                </field>
+                <group position="inside">
+                        <filter name="group_product_id" string="Product"
+                                domain="[]"
+                                context="{'group_by':'product_id'}"/>
+                </group>
+            </field>
+        </record>
+
+        <record id="view_move_form" model="ir.ui.view">
+            <field name="name">account.move.form</field>
+            <field name="model">account.move</field>
+            <field name="inherit_id" ref="account.view_move_form"/>
+            <field name="arch" type="xml">
+                <xpath expr="//field[@name='line_ids']/tree//field[@name='partner_id']" position="after">
```

Comment:
@lreficent You usually put the full path to avoid duplicate entries found in the search. As the field is in a tree view, another module could add the same field in a form view, then you'd have the same field 2 times, and the application would complain.